AirCasting/Services/OldMeasurementRemovers/DefaultRemoveOldMeasurementsService.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+import CoreData
+import Resolver
+
+protocol RemoveOldMeasurements {
+    func removeOldestMeasurements(in context: NSManagedObjectContext, from sessionUUID: SessionUUID) throws
+}
+
+final class DefaultRemoveOldMeasurementsService: RemoveOldMeasurements {
+    
+    enum Error: Swift.Error {
+        case removingMeasurementsSurplusFailed
+        case noStreamInSession
+    }
+    
+    @Injected private var measurementStreamStorage: MeasurementStreamStorage
+    
+    /// In fixed and external sessions we need to remove measurements older than 24h, and we treat 24 like a date.
+    /// We know the time of the last measurement and based on that we are subtracking 24h in seconds.
+    func removeOldestMeasurements(in context: NSManagedObjectContext, from sessionUUID: SessionUUID) throws {
+        if let session = try? context.existingSession(uuid: sessionUUID) {
+            try removeMeasurementsSurplus(context: context, streams: session.allStreams)
+        } else {
+            let session = try context.existingExternalSession(uuid: sessionUUID)
+            try removeMeasurementsSurplus(context: context, streams: session.allStreams)
+        }
+    }
+    
+    private func removeMeasurementsSurplus(context: NSManagedObjectContext, streams: [MeasurementStreamEntity]) throws {
+        for stream in streams {
+            timeBasedRemover(context: context, stream: stream)
+        }
+    }
+    
+    private func timeBasedRemover(context: NSManagedObjectContext, stream: MeasurementStreamEntity) {
+        guard let lastMeasurementTime = stream.allMeasurements?.last?.time else { Log.error("No last measurement when trying to remove those from > 24h"); return }
+        let threshold = lastMeasurementTime.twentyFourHoursBefore
+        let result = context.deleteMeasurements(thresholdInMiliseconds: threshold, stream: stream)
+        switch result {
+        case .success(let success):
+            try! context.save()
+            Log.info(success)
+        case .failure(let error):
+            Log.error(error.localizedDescription)
+        }
+    }
+}
+
+extension Date {
+    var twentyFourHoursBefore: Double {
```

Comment:
I think we can actually return it as a date, and then we wouldn't have to change it back to Date in deleteMeasurements function. It would look like this:
```
var twentyFourHoursBefore: Date {
        self - (24 * 60 * 60)
    }
```

Reply:
Also, I think it can be added in the Date+Extensions file

Reply:
I moved this extension to the file but I prefer to stay with miliseconds approach so then we are able to easly and in clear way use NSDate: ` NSDate(timeIntervalSince1970: thresholdInMiliseconds/1000)`

Reply:
Then I'd suggest adding inMiliseconds to the name for consistency, but I'll leave it up to you 🙂